In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import pandas as pd
import numpy as np
from sklearn import preprocessing
import seaborn as sns
import matplotlib as mpl
import matplotlib.pyplot as plt
%matplotlib inline

In [3]:
!pip install quilt
!quilt install ResidentMario/missingno_data

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 110 kB 10.6 MB/s 
100% 420k/420k [00:00<00:00, 653kB/s] 


In [4]:
!pip install lxml
!pip install beautifulsoup4

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [5]:
!pip install chardet

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [6]:
from quilt.data.ResidentMario import missingno_data

In [7]:
collisions = missingno_data.nyc_collision_factors()
collisions = collisions.replace("nan", np.nan)

In [8]:
conteudo = pd.read_csv('/content/drive/My Drive/TCC/csv/conteudo_das_anotacoes.csv', encoding="utf-8")
lugar = pd.read_csv('/content/drive/My Drive/TCC/csv/lugar_das_anotacoes.csv')
ativ = pd.read_csv('/content/drive/My Drive/TCC/csv/nome_das_atividades.csv')
onto = pd.read_csv('/content/drive/My Drive/TCC/csv/ontologias.csv')
titulo = pd.read_csv('/content/drive/My Drive/TCC/csv/titulo_obras_literarias.csv')

In [9]:
import chardet 

asdas = chardet.detect(conteudo['content'][13].encode())
print(asdas)

{'encoding': 'utf-8', 'confidence': 0.99, 'language': ''}


In [37]:
teste = conteudo['content'][11]
tst1 = teste.encode('ascii').decode('UTF-8')
tst1

'<p>\n\tProcure por informa&ccedil;&otilde;es sobre a ret&oacute;rica cl&aacute;ssica. Veja, por exemplo: Barthes, Roland. &quot;A Ret&oacute;rica Antiga&quot; in <em>Pesquisas de Ret&oacute;rica</em>, Petr&oacute;polis: Vozes, 1975.</p>\n'

In [38]:
from bs4 import BeautifulSoup

# Utilizando o beautiful soup para pegar o texto que esta dentro do html da anotacao
soup = BeautifulSoup(teste, 'lxml')

tst2 = soup.get_text()
# TODO: usar regex para formatar as quebras de linhas ou pesquisar se o bs4 faz isso
tst2

'\n\tProcure por informações sobre a retórica clássica. Veja, por exemplo: Barthes, Roland. "A Retórica Antiga" in Pesquisas de Retórica, Petrópolis: Vozes, 1975.\n'

In [12]:
conteudo.head()

ref_annotation  type                                 title  \
0               2     0                               Títulos   
1               3     6                             Evangelho   
2               7     4  O valor da ação e o valor da palavra   
3               8     4  O valor da ação e o valor da palavra   
4               9     4  O valor da ação e o valor da palavra   

                                             content  
0  Nunca deixe de examinar com atenção os títulos...  
1                     Leia a "Parábola do semeador".  
2                                                NaN  
3     Missão dos jesuítas (como no parágrafo acima).  
4  Uma retórica da metalinguagem (defesa da autor...

In [13]:
conteudo['type'].count()

82360

In [14]:
conteudo['type'].unique()

array([0, 6, 4, 5, 1, 3, 2])

In [15]:
conteudo['type'].describe

<bound method NDFrame.describe of 0        0
1        6
2        4
3        4
4        4
        ..
82355    0
82356    0
82357    0
82358    0
82359    0
Name: type, Length: 82360, dtype: int64>

In [16]:
conteudo['type'].isnull().sum()

0

In [17]:
conteudo.where(conteudo['type'] == 0)

ref_annotation  type    title  \
0                 2.0   0.0  Títulos   
1                 NaN   NaN      NaN   
2                 NaN   NaN      NaN   
3                 NaN   NaN      NaN   
4                 NaN   NaN      NaN   
...               ...   ...      ...   
82355        219722.0   0.0      NaN   
82356        219725.0   0.0      NaN   
82357        219728.0   0.0      NaN   
82358        219731.0   0.0      NaN   
82359        219732.0   0.0      NaN   

                                                 content  
0      Nunca deixe de examinar com atenção os títulos...  
1                                                    NaN  
2                                                    NaN  
3                                                    NaN  
4                                                    NaN  
...                                                  ...  
82355  <p>A plateia [...] n&atilde;o compreende a fal...  
82356  <p>No sonho o al&iacute;vio. A presen&ccedil;a...  
82357       <p>Sem as duas almas, perde-se a vida.</p>\n  
82358  <p>Receio de achar-se um ou dois ao mesmo temp...  
82359                          <p>O gesto mutilado</p>\n  

[82360 rows x 4 columns]

In [18]:
conteudo['content'].isnull().sum()

1

In [19]:
anots =  conteudo['content'].dropna()

In [20]:
from ast import literal_eval

In [21]:
# z = conteudo['content'].dropna().apply(lambda x: literal_eval(x).decode('UTF-8'))

In [34]:
def clean_anot(anots):
  new = []
  for anot  in anots:        
    # anot = anot.encode()  
    x = anot.encode('utf-8').decode('utf-8')
    # print(anot.encode('utf_8'))
    # print(type(anot))
    # print('printou')
    new.append(x)
  return new
  



In [35]:
x = clean_anot(conteudo['content'].dropna())

In [36]:
x

['Nunca deixe de examinar com atenção os títulos e os nomes. Você procurou o que quer dizer exatamente "sexagésima"?',
 'Leia a "Parábola do semeador".',
 'Missão dos jesuítas (como no parágrafo acima).',
 'Uma retórica da metalinguagem (defesa da autoridade no discurso).',
 'Uma retórica da metalinguagem (defesa da autoridade no discurso).',
 'Valor específico, atual, atual e moral da Escritura.',
 'Valor específico, atual e moral da Escritura.',
 'Veja o 3º, o 4 e o 5º parágrafos da parte I.',
 'Valor específico, moral e atual da Escritura.',
 'Valor específicos, moral e atual da escritura',
 '<p>\n\tProcure por informa&ccedil;&otilde;es sobre a ret&oacute;rica cl&aacute;ssica. Veja, por exemplo: Barthes, Roland. &quot;A Ret&oacute;rica Antiga&quot; in <em>Pesquisas de Ret&oacute;rica</em>, Petr&oacute;polis: Vozes, 1975.</p>\n',
 'Em que elemento da retórica clássica este trecho se fundamenta? E veja se isso condiz, de fato, com a própria linguagem de Vieira.',
 'Aqui, Vieira critic

In [28]:
x = conteudo['content'].apply(lambda x: x.encode('unicode_escape').decode('UTF-8'))

AttributeError: ignored

In [29]:
x = conteudo['content'].apply(lambda x: x.encode('UTF-8').
                 decode('UTF-8') if isinstance(x, str) else x)

In [30]:
x

0        Nunca deixe de examinar com atenção os títulos...
1                           Leia a "Parábola do semeador".
2                                                      NaN
3           Missão dos jesuítas (como no parágrafo acima).
4        Uma retórica da metalinguagem (defesa da autor...
                               ...                        
82355    <p>A plateia [...] n&atilde;o compreende a fal...
82356    <p>No sonho o al&iacute;vio. A presen&ccedil;a...
82357         <p>Sem as duas almas, perde-se a vida.</p>\n
82358    <p>Receio de achar-se um ou dois ao mesmo temp...
82359                            <p>O gesto mutilado</p>\n
Name: content, Length: 82360, dtype: object

In [ ]:
type(conteudo['content'][0])

In [ ]:
conteudo['content'].isna().sum()

Temos que 53748 titulos das anotações são nulos de um total de 82360 anotações,
temos então que aproximadamente 65% das anotações não tem titulo.



In [ ]:
# Podemos tenta verificar se as anotações que contém titulo se saem melhor que as que não tem
conteudo['title'].isna().sum()

In [ ]:
len(conteudo['title'])

In [ ]:
len(conteudo['title'].unique())

Temos 82360 conteudos de anotações, sendo que 78808 são unicos. É preciso se atentar com os que são repetidos, fazer a avaliação destes.


In [ ]:
len(conteudo['content'].unique())

In [ ]:
len(conteudo['content'])

Grafico que contém a quantidade de cada tipo de anotação. Podemos notar que há um grande desbalanceamento, sendo a do tipo 0 com a maior quantidade

In [ ]:
plt.hist(conteudo['type'], bins=20)
plt.figure(figsize=(15,10))
plt.show()

Filtrando o conjunto de dados para que se retire o tipo 0. Mesmo retirando o tipo 0, podemos notar que ainda a diferença entre os tipos é bem grande.


In [ ]:
tp = conteudo['type'].where(conteudo['type']!=0).dropna()

In [ ]:
plt.hist(tp, bins=20)
plt.figure(figsize=(15,10))
plt.show()

In [ ]:
lugar.columns

In [ ]:
ativ.columns

In [ ]:
onto.columns

In [ ]:
titulo.columns

In [ ]:
ativ.head()